r/PACKAGING.md

```diff
@@ -26,6 +26,7 @@ For a high-level overview of the release process see the
 ## Before the release candidate is cut
 
 - [ ] [Create a GitHub issue](https://github.com/apache/arrow/issues/new/) entitled `[R] CRAN packaging checklist for version X.X.X` and copy this checklist to the issue.
+- [ ] Review deprecated functions to advance their deprecation status, including removing preprocessor directives that no longer apply (e.g., oudated R versions, Arrow C++ versions older than the last three major versions).
```

Comment:
I'd suggest being more concrete about what to review. For example, search for `#if ARROW_VERSION_MAJOR` (and whatever the R version one is) in `r/src`.